PID-Analyzer.py

```diff
@@ -513,71 +512,79 @@ def beheader(self, loglist):
+            path_root, path_ext = os.path.splitext(os.path.basename(fpath))
+            temp_path = os.path.join(
+                self.tmp_dir, '%s_temp%d%s' % (path_root, i, path_ext))
+            with open(temp_path, 'wb') as newfile:
+                newfile.write(firstline+split[i])
+            bbl_sessions.append(temp_path)
 
         loglist = []
-        for t in temps:
-            size = os.path.getsize(self.path+'/'+t)
-            if size>500000:
+        for bbl_session in bbl_sessions:
+            size_bytes = os.path.getsize(os.path.join(self.tmp_dir, bbl_session))
+            if size_bytes > LOG_MIN_BYTES:
                 try:
-                    msg = os.system('blackbox_decode.exe' + ' '+t)
-                    loglist.append(t)
+                    msg = os.system(self.blackbox_decode_bin_path + ' '+bbl_session)
+                    loglist.append(bbl_session)
                 except:
-                    print 'Error in Blackbox_decode'
+                    logging.error('Error in Blackbox_decode of %r' % bbl_session)
             else:
-                os.remove(t)
+                # There is often a small bogus session at the start of the file.
+                logging.warning(
+                    'Ignoring BBL session %r, %dB < %dB.'
+                    % (bbl_session, size_bytes, LOG_MIN_BYTES))
+                os.remove(bbl_session)
         return loglist
 
 
-
-def main():
-    ### use here via:
-    #test = BB_log('path/file.bbl', 'addidtional name')
-    #plt.show()
-
-    print Version +'\n\n'
-    print 'Hello Pilot!'
-    print 'This program uses Blackbox_decode:\n' \
-          'https://github.com/cleanflight/blackbox-tools/releases\n' \
-          'to generate .csv files from your log.\n' \
-          'Please put logfiles, Blackbox_decode.exe and this program into a single folder.\n'
+def main(log_file_path, plot_name, blackbox_decode):
+    logging.info(Version)
+    logging.info('Hello Pilot!')
 
     while True:
-        file = raw_input("Place your log here: \n-->\n")
-        name = raw_input('\nName for this plot: (optional)\n')
-        #thr_mode = raw_input('\n Sort respose by throttle? (experimental) Press y!')
-        test = BB_log(str(file), str(name))
+        test = BB_log(log_file_path, plot_name, blackbox_decode)
         plt.show()
 
+
+def strip_quotes(filepath):
+    """Strips single or double quotes from a string."""
+    if filepath[-1]=='"' or filepath[-1]=="'":
+        return filepath[1:-1]
+    else:
+        return filepath
+
+
 if __name__ == "__main__":
-    main()
+    logging.basicConfig(
+        format='%(levelname)s %(asctime)s %(filename)s:%(lineno)s: %(message)s',
+        level=logging.INFO)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('log', help='BBL log file to analyse.')
+    parser.add_argument('-n', '--name', default='', help='Plot name.')
+    parser.add_argument(
+        '--blackbox_decode', default='./Blackbox_decode.exe',
```

Comment:
Found that on my win10 system the `./Blackbox_decode.exe` does not work, but callig just `Blackbox_decode.exe` (without the `./` does the job. What are you using?

Reply:
I'm using MacOS X, so it makes sense that the `/` works for me but wouldn't work on Windows. Fixed to be portable.